docker/docker_client.go

```diff
@@ -265,7 +265,12 @@ func (c *dockerClient) setupRequestAuth(req *http.Request) error {
 		}
 		chs := parseAuthHeader(res.Header)
 		if res.StatusCode != http.StatusUnauthorized || chs == nil || len(chs) == 0 {
-			// try again one last time with Basic Auth (gcr.io for instance)
+			if res.StatusCode == http.StatusOK {
```

Comment:
Is it really just `StatusOK`? What about `StatusNotFound`, `StatusAccepted` (both used in `deleteImage` and `PutBlob`), `StatusCreated` (`PutBlob`, `Putmanifest`)?

Or perhaps this should really be `if res.StatusCode != http.StatusUnauthorized`? I don’t readily have a reproducer for #191 so I don’t know which of the conditions should get us into the `testReq2` path.